types/src/v0/v0_3/stake_table.rs

```diff
@@ -103,3 +103,79 @@ pub enum StakeTableEvent {
     KeyUpdate(ConsensusKeysUpdated),
     KeyUpdateV2(ConsensusKeysUpdatedV2),
 }
+
+
+#[derive(Debug, Error)]
+pub enum StakeTableError {
```

Comment:
Sorry this is going to blow up a bit. 

I think we need to distinguish 2 types of errors. One type are errors that are expected because the contract allows it. For example unauthenticated events, or (currently) duplicate schnorr keys. These are errors we need to handle correctly and continue (ignoring the offending events).

The other types of errors indicate a bug in the contract. They should never happen. If they do happen the GCL nodes should not continue running at all.

Reply:
Correction, since we pre-filter unauthenticated events that is also an unexpected error. But duplicate schnorr keys are possible with the current stake table contract.

Reply:
https://github.com/EspressoSystems/espresso-network/pull/3396/commits/ba027c88b5fcc7226a6afaa0557b0e401058f097